### PR TITLE
Branch tab daily

### LINKED ISSUE #72: DAILY tab - Coding - Allows user to add todo tasks from existing tasks

### LINKED ISSUE #73: DAILY tab - UI - Showing todo tasks which are added

### PR DESCRIPTION
**Pull request by yingqi0607**

### CODE REVIEW

File: src/main/java/seedu/tr4cker/logic/commands/DailyCommand.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) { // short circuit if same object
+            return true;
+        } else if (!(other instanceof DailyCommand)) {
+            return false;
+        } else {
+            return false; // to be implemented
```
Comment (ethan-l-m-e) [inline src/main/java/seedu/tr4cker/logic/commands/DailyCommand.java]: Did u mean to leave it as is?

File: src/main/java/seedu/tr4cker/logic/commands/TodoCommand.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import seedu.tr4cker.commons.core.Messages;
+import seedu.tr4cker.commons.core.index.Index;
+import seedu.tr4cker.logic.commands.exceptions.CommandException;
+import seedu.tr4cker.model.Model;
+import seedu.tr4cker.model.daily.Todo;
+import seedu.tr4cker.model.task.Deadline;
+import seedu.tr4cker.model.task.Name;
+import seedu.tr4cker.model.task.Task;
+
+
```
Comment (ethan-l-m-e) [inline src/main/java/seedu/tr4cker/logic/commands/TodoCommand.java]: Missing Javadoc here

File: src/main/java/seedu/tr4cker/logic/parser/DailyCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import seedu.tr4cker.logic.parser.exceptions.ParseException;
+
+
+/**
+ * Parses input arguments and creates a new DailyCommand object.
+ */
+public class DailyCommandParser implements Parser<DailyCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the CountdownCommand
+     * and returns an CountdownCommand object for execution.
```
Comment (ethan-l-m-e) [inline src/main/java/seedu/tr4cker/logic/parser/DailyCommandParser.java]: I think its supposed to be DailyCommand object?

File: src/main/java/seedu/tr4cker/model/daily/Todo.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Todo {
+    private final Name name;
+    private final Deadline deadline;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Todo(Name name, Deadline deadline) {
```
Comment (ethan-l-m-e) [inline src/main/java/seedu/tr4cker/model/daily/Todo.java:22]: Idk,  might need to _requireAllNonNull()_?

File: src/main/java/seedu/tr4cker/logic/commands/TodoCommand.java (hunk 2 above)
Comment (ruilingk) [inline src/main/java/seedu/tr4cker/logic/commands/TodoCommand.java]: Remember to add Javadocs!

File: src/main/java/seedu/tr4cker/model/daily/Todo.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        this.name = name;
+        this.deadline = deadline;
+    }
+
+    public Name getName() {
```
Comment (ruilingk) [inline src/main/java/seedu/tr4cker/model/daily/Todo.java:31]: Javadocs for public methods.

File: src/main/java/seedu/tr4cker/model/daily/UniqueDailyList.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+     * The todo must exist in the list.
+     */
+    public void remove(Todo toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new TodoNotFoundException();
+        }
+    }
+
+    public void setTodos(UniqueDailyList replacement) {
```
Comment (ruilingk) [inline src/main/java/seedu/tr4cker/model/daily/UniqueDailyList.java:67]: Bump!

File: src/main/java/seedu/tr4cker/storage/JsonSerializableTr4cker.java [hunk 7]
```diff
@@ -105,6 +126,8 @@ public Tr4cker toModelType() throws IllegalValueException {
         modulesToModelType(tr4cker);
         tasksToModelType(tr4cker);
         eventsToModelType(tr4cker);
+        modulesToModelType(tr4cker);
```
Comment (ethan-l-m-e) [inline src/main/java/seedu/tr4cker/storage/JsonSerializableTr4cker.java]: duplicate _modulesToModelType()_, now there are double calls?

Keep the one at the top.

**Review by ruilingk — COMMENTED** [review]
Some minor addition of Javadocs and it LGTM.

File: src/main/java/seedu/tr4cker/ui/daily/DailyTabWindow.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+     * @param logic Logic of TR4CKER.
+     */
+    public DailyTabWindow(Logic logic) {
+        super(FXML);
+        logger.fine("Initialising Daily Tab Window...");
+        this.dailyPanel = new DailyPanel(logic.getFilteredTodoList());
+        dailyPanelPlaceholder.getChildren().add(dailyPanel.getRoot());
+        logger.fine("Created Daily Tab Window.");
+    }
+}
```
Comment (ethan-l-m-e) [inline src/main/java/seedu/tr4cker/ui/daily/DailyTabWindow.java:33]: Not sure if u got EOF here. In TodoCard and your fxml files as well.

**Review by ruilingk — APPROVED** [review]
lgtm!!

**Review by ethan-l-m-e — APPROVED** [review]
lgmt

**Pull request merged**